presto-main/src/main/java/com/facebook/presto/server/remotetask/WorkerTaskStatusFetcher.java

```diff
@@ -0,0 +1,284 @@
+
+        this.errorTracker = new RequestErrorTracker(taskId, initialTaskStatus.getSelf(), maxErrorDuration, errorScheduledExecutor, "getting task status");
+        this.stats = requireNonNull(stats, "stats is null");
+        this.isBinaryTransportEnabled = isBinaryTransportEnabled;
+    }
+
+    public void addTask(TaskId taskId, ContinuousBatchTaskStatusFetcher.Task task) {
+        idTaskMap.put(taskId, task);
+    }
+
+    TaskStatus getTaskStatus(TaskId taskId)
+    {
+        return idTaskMap.get(taskId).taskStatus.get();
+    }
+
+    // TODO: Schedule request per worker
+    public synchronized void scheduleNextRequest()
+    {
+        // stopped or done?
+        if (!running) {
+            return;
+        }
+
+        // outstanding request?
+        if (future != null && !future.isDone()) {
+            // this should never happen
+            log.error("Can not reschedule update because an update is already running");
+            return;
+        }
+
+        // if throttled due to error, asynchronously wait for timeout and try again
+        ListenableFuture<?> errorRateLimit = errorTracker.acquireRequestPermit();
+        if (!errorRateLimit.isDone()) {
+            errorRateLimit.addListener(this::scheduleNextRequest, executor);
+            return;
+        }
+
+        Request request = setContentTypeHeaders(isBinaryTransportEnabled, prepareGet())
+                .setUri(uriBuilderFrom(worker.getSelf()).appendPath("all_status").build())
+                .setHeader(PRESTO_CURRENT_STATE, worker.getState().toString())
+                .setHeader(PRESTO_MAX_WAIT, refreshMaxWait.toString())
+                .build();
+
+        ResponseHandler responseHandler;
+        if (isBinaryTransportEnabled) {
+            responseHandler = createFullSmileResponseHandler((SmileCodec<List<TaskStatus>>) taskListStatusCodec);
+        }
+        else {
+            responseHandler = createAdaptingJsonResponseHandler(unwrapJsonCodec(taskListStatusCodec));
+        }
+
+        errorTracker.startRequest();
+        future = httpClient.executeAsync(request, responseHandler);
+        currentRequestStartNanos.set(System.nanoTime());
+        Futures.addCallback(future, new SimpleHttpResponseHandler<>(this, request.getUri(), stats), executor);
+    }
+
+    @Override
+    public void success(List<TaskStatus> value)
+    {
+        try (SetThreadName ignored = new SetThreadName("ContinuousBatchTaskStatusFetcher-%s", idTaskMap)) {
+            updateStats(currentRequestStartNanos.get());
+            try {
+                updateAllTaskStatus(value);
+                errorTracker.requestSucceeded();
+            }
+            finally {
+                // Signal to CBTSF we're done
+            }
+        }
+    }
+
+    @Override
+    public void failed(Throwable cause)
+    {
+        try (SetThreadName ignored = new SetThreadName("ContinuousTaskStatusFetcher-%s", idTaskMap)) {
```

Comment:
just name the thread with taskId only no need to pass the map here. i.e. replace idTaskMap to just taskId.

Reply:
Since this is on a per worker basis, I wanted to account for all tasks running on the worker (though theoretically we could get one task and use that as the thread name, which isn't as pretty).

Reply:
yeah we should name it with worker name to easily identify them in this case.